cardano-db/src/Cardano/Db/Insert.hs

```diff
@@ -47,6 +47,12 @@ insertTxIn = insertByReturnKey
 insertTxOut :: MonadIO m => TxOut -> ReaderT SqlBackend m TxOutId
 insertTxOut = insertByReturnKey
 
+insertPoolOnData
```

Comment:
I think we need to carefully separate Shelley specific features from Byron especially since the two chains are represetned by different types.

Reply:
Sure, but I would argue that it's still early for that.
The way I see it in phases:
1. add all function in the same module
2. you see duplication in the module and you create a separate module for some
3. you see more duplication and you abstract over both modules